v2/tools/generator/internal/astmodel/errored_type.go

```diff
@@ -130,22 +129,26 @@ func (e *ErroredType) RequiredPackageReferences() *PackageReferenceSet {
 }
 
 func (e *ErroredType) handleWarningsAndErrors() {
-	for _, warning := range e.warnings {
-		klog.Warning(warning)
-	}
+	var errs []error
 
 	if len(e.errors) > 0 {
-		var errs []error
 		for _, err := range e.errors {
 			errs = append(errs, errors.New(err))
 		}
+	}
 
-		if len(errs) == 1 {
-			panic(errs[0])
-		} else {
-			panic(kerrors.NewAggregate(errs))
+	// Treating warnings as errors isn't quite right, but good enough for now
```

Comment:
I think this might be bad in a lot of cases. It's a contradiction between "Oh! We're not expecting this, but we'll take it" and "Oh! something is wrong/broken and we can not proceed with it"

Reply:
I agree in principle - but I don't thinkg we get any warnings at the moment anyway. I've got a note on my TODO list to tidy this up.

Reply:
I'll set up the pipeline to check for warnings and emit them just prior to code generation.